bedrock/firefox/tests/test_views.py

```diff
@@ -555,6 +545,41 @@ def test_experiment_redirect_automation_param(self, render_mock):
         assert 'cache-control' not in resp
 
 
+class TestFirefoxYandex(TestCase):
+    @patch.dict(os.environ, SWITCH_FIREFOX_YANDEX='True')
+    @patch.object(views, 'ftl_file_is_active', lambda *x: True)
+    def test_yandex_show_to_ru(self):
```

Comment:
@pmac I've progressed to a failure because my request is returning the wrong content. Progress! 

The content is only on the page if the local is `ru`. If I look at the doc being the `lang` attribute on the `<html>` element is `en`... but the copy is in Russian. Not sure if that's a symptom of a different oddity.

Reply:
I think it's because you're switching things in the template based on the `LANG` variable, which is different from the `request.lang` property. You should be able to activate a language with:

```python
with self.activate('ru'):
    response = view(req)
```

Replace line 555 with that and it may work. Your other option is to not make these conditionals in the template, and instead make a new yandex template that inherits from this one and that overrides the bits that you want. It'd be a bit easier to test I'm guessing.